From b57b7d0ecb8e0089a6a93faca2fc852a1dcad8f8 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sjpark@amazon.de>
Date: Mon, 31 May 2021 13:28:16 +0000
Subject: [RFC PATCH 00/13] *** SUBJECT HERE ***
To: akpm@linux-foundation.org
Cc: Jonathan.Cameron@Huawei.com,
    acme@kernel.org,
    alexander.shishkin@linux.intel.com,
    amit@kernel.org,
    benh@kernel.crashing.org,
    brendanhiggins@google.com,
    corbet@lwn.net,
    david@redhat.com,
    dwmw@amazon.com,
    elver@google.com,
    fan.du@intel.com,
    foersleo@amazon.de,
    greg@kroah.com,
    gthelen@google.com,
    guoju.fgj@alibaba-inc.com,
    mgorman@suse.de,
    minchan@kernel.org,
    mingo@redhat.com,
    namhyung@kernel.org,
    peterz@infradead.org,
    riel@surriel.com,
    rientjes@google.com,
    rostedt@goodmis.org,
    rppt@kernel.org,
    shakeelb@google.com,
    shuah@kernel.org,
    sj38.park@gmail.com,
    snu@amazon.de,
    vbabka@suse.cz,
    vdavydov.dev@gmail.com,
    zgf574564920@gmail.com,
    linux-damon@amazon.com,
    linux-mm@kvack.org,
    linux-doc@vger.kernel.org,
    linux-kernel@vger.kernel.org

*** BLURB HERE ***

SeongJae Park (13):
  mm/damon/paddr: Support the pageout scheme
  mm/damon/damos: Make schemes aggressiveness controllable
  damon/core/schemes: Skip already charged targets and regions
  mm/damon/dbgfs: Support schemes speed limit
  mm/damon/selftests: Support schemes speed limit
  mm/damon/schemes: Prioritize regions within speed limit
  mm/damon/vaddr,paddr: Support pageout prioritization
  mm/damon/dbgfs: Support prioritization weights
  tools/selftests/damon: Update for regions prioritization of schemes
  mm/damon/schemes: Activate schemes based on a watermarks mechanism
  mm/damon/dbgfs: Support watermarks
  selftests/damon: Support watermarks
  mm/damon: Introduce DAMON-based reclamation

 include/linux/damon.h                         | 118 ++++++++-
 mm/damon/Kconfig                              | 128 ++++++++++
 mm/damon/Makefile                             |   1 +
 mm/damon/core.c                               | 215 +++++++++++++++-
 mm/damon/dbgfs.c                              |  45 +++-
 mm/damon/paddr.c                              |  52 +++-
 mm/damon/prmtv-common.c                       |  48 +++-
 mm/damon/prmtv-common.h                       |   5 +
 mm/damon/reclaim.c                            | 230 ++++++++++++++++++
 mm/damon/vaddr.c                              |  15 ++
 .../testing/selftests/damon/debugfs_attrs.sh  |   4 +-
 11 files changed, 834 insertions(+), 27 deletions(-)
 create mode 100644 mm/damon/reclaim.c

-- 
2.17.1

